New IA - Overview pages (#5700)
TykTechnologies/tyk-docs: tyk-docs/content/tyk-components.md

---

# Understanding Tyk’s API Management Components

Tyk offers a full ecosystem of API management tools, supporting every stage of the API lifecycle—from development and deployment to monitoring and scaling. With Tyk’s components, teams can seamlessly integrate, secure, and scale APIs across different environments, facilitating robust and reliable API operations.

---

## Tyk’s API Management Components Overview

Below is a detailed introduction to each of Tyk’s components and their individual roles within an API management strategy. For more information, follow the links provided to explore specific documentation on each component.

{{< grid >}}

{{< badge title="Tyk Gateway" href="tyk-components#tyk-gateway" >}} The open-source core of Tyk that handles request routing, rate-limiting, and caching. {{< /badge >}}

{{< badge title="Tyk Dashboard" href="tyk-components#tyk-dashboard" >}} A powerful interface for managing and monitoring APIs, users, and permissions. {{< /badge >}}

{{< badge title="Tyk Enterprise Developer Portal" href="tyk-components#tyk-enterprise-developer-portal" >}} A customizable portal for enabling API discovery and user self-service. {{< /badge >}}

> {{< badge title="Tyk Multi Data Centre Bridge" href="tyk-components#tyk-multi-data-centre-bridge-mdcb" >}} Ensures high availability by synchronizing APIs across multiple data centers. {{< /badge >}}

Reviewer: ```suggestion
{{< badge title="Tyk Multi Data Centre Bridge" href="tyk-components#tyk-multi-data-centre-bridge-mdcb" >}} Centralized API management across distributed data centers for global reach and compliance. {{< /badge >}}
```

{{< badge title="Tyk Pump" href="tyk-components#tyk-pump" >}} Pushes analytics data from the Gateway to external data stores for advanced insights. {{< /badge >}}

{{< badge title="Tyk Operator" href="tyk-components#tyk-operator" >}} Integrates with Kubernetes for seamless API management within containerized environments. {{< /badge >}}

{{< badge title="Tyk Streams" href="tyk-components#tyk-streams" >}} Enables real-time event streaming for APIs, ideal for push notifications and live data. {{< /badge >}}

> {{< badge title="Tyk Sync" href="tyk-components#tyk-sync" >}} Automates the synchronization of API configurations across environments. {{< /badge >}}

Reviewer: ```suggestion
{{< badge title="Tyk Sync" href="tyk-components#tyk-sync" >}} Synchronizes API configurations across environments for consistency and streamlined deployments. {{< /badge >}}
```

- **Community Engagement**: Features like FAQs and forums foster a collaborative developer community.

The Developer Portal streamlines API adoption by offering developers a comprehensive platform for API discovery and self-service.

---

### Tyk Multi Data Centre Bridge (MDCB)

The **[Tyk Multi Data Centre Bridge (MDCB)]({{< ref "tyk-multi-data-centre" >}})** enables centralized control over API configurations across distributed regions or data centers. It’s ideal for global API infrastructure, offering:

- **Centralized Management**: Controls API configurations across different regions from a single interface.
- **Data Isolation**: Supports regulatory compliance by keeping data within specified regions.
- **Enhanced Resilience**: Ensures high availability for distributed API deployments.

MDCB is designed for organizations with extensive geographic reach, supporting both global distribution and local compliance.

---

### Tyk Pump

**[Tyk Pump]({{< ref "tyk-pump" >}})** is a lightweight tool that aggregates API analytics from the Tyk Gateway and exports them to various storage systems for analysis:

- **Data Collection**: Continuously gathers real-time API analytics.

> - **Data Export**: Integrates with multiple backends like MongoDB, AWS S3, and others.

Reviewer: ```suggestion
- **Data Export**: Integrates with multiple backends like Splunk, Prometheus, ElasticSearch, and others.
```

- **Version Control**: Manages versioning and consistency across deployments.
- **Standardization**: Ensures uniform setups across environments, improving reliability.

Tyk Helm Charts are particularly useful for teams deploying Tyk in Kubernetes, streamlining installation and ensuring consistent configurations.

---

### Universal Data Graph (UDG)

The **[Universal Data Graph (UDG)]({{< ref "universal-data-graph" >}})** provides a single API endpoint for accessing data from multiple sources using GraphQL, offering:

- **Data Aggregation**: Combines data from disparate sources into a unified API.
- **Customizable Data Access**: Clients can query only the data they need.
- **Secure Data Management**: Implements access controls to manage data permissions.

The UDG simplifies complex data interactions, enabling developers to work with a unified GraphQL endpoint for various backend services.

---

## How Tyk’s Components Work Together

Tyk’s components create a unified API management ecosystem, covering all aspects of API lifecycle management:

- **Traffic and Security**: The Tyk Gateway acts as the first line of defense, handling traffic and enforcing security protocols.

> - **Configuration and Deployment**: Tyk Dashboard, Tyk Operator, and Tyk Helm Charts provide tools for managing, deploying, and scaling APIs.

Reviewer: ```suggestion
- **Configuration and Deployment**: Tyk Dashboard, Tyk Operator, and Tyk Sync provide tools for managing, deploying, and scaling APIs.
```

> - **Developer Experience**: The Tyk Developer Portal enhances the developer experience, promoting API discovery and user engagement.

Reviewer: ```suggestion
- **Developer Experience**: The Tyk Developer Portal and Identity Broker enhances the developer experience, promoting API discovery and user engagement.
```

> - **Global Reach and Resilience**: MDCB and Tyk Sync ensure consistent API configurations and high availability across regions.

Reviewer: ```suggestion
- **Global Reach and Resilience**: MDCB ensures consistent API configurations and high availability across regions.
```

> - **Data and Insights**: Tyk Pump and Tyk Streams support real-time monitoring and data streaming, enabling data-driven decisions.

Reviewer: ```suggestion
- **Data and Insights**: Tyk Gateway's native support for OpenTelemetry and Tyk Pump support real-time monitoring and analytics.
```

> - **Unified Access**: UDG and Identity Broker provide unified data and identity management, simplifying complex API interactions.

Reviewer: ```suggestion
- **Unified Access**: UDG and Tyk Streams unify API Management by enabling seamless, multi-protocol data integration and delivery.
- **Kubernetes Deployment**: Tyk Helm Charts simplifies deployment of Tyk on Kubernetes.
```
